src/samples/perf-benchmarks/Program.cs

```diff
@@ -7,52 +7,127 @@
-    await remotePeer.DialAsync<PerfProtocol>();
-    TimeSpan elapsed = timeSpent.Elapsed;
-    Console.WriteLine("Libp2p");
-    Console.WriteLine("Elapsed {0}", timeSpent.Elapsed);
-    Console.WriteLine("Speed {0:0.00} MiB/s", PerfProtocol.TotalLoad / timeSpent.Elapsed.TotalMilliseconds * 1000 / 1024 / 1024);
-    await remotePeer.DisconnectAsync();
-}
-await Task.Delay(1000);
+        // Creating peers for communication
+        ILocalPeer localPeer = peerFactory.Create();
+        ILocalPeer remotePeer = peerFactory.Create();
 
-{
-    IPeerFactory peerFactory = NoStackPeerFactoryBuilder.Create
-        .AddAppLayerProtocol<PerfProtocol>()
-        .Build();
-
-    ILocalPeer peer = peerFactory.Create();
-    IListener listener = await peer.ListenAsync($"/ip4/0.0.0.0/tcp/0");
-
-    Multiaddress remoteAddr = listener.Address;
-    ILocalPeer localPeer = peerFactory.Create();
-    IRemotePeer remotePeer = await localPeer.DialAsync(remoteAddr);
-
-    Stopwatch timeSpent = Stopwatch.StartNew();
-    await remotePeer.DialAsync<PerfProtocol>();
-    TimeSpan elapsed = timeSpent.Elapsed;
-    Console.WriteLine("NoStack");
-    Console.WriteLine("Elapsed {0}", timeSpent.Elapsed);
-    Console.WriteLine("Speed {0:0.00} MiB/s", PerfProtocol.TotalLoad / timeSpent.Elapsed.TotalMilliseconds * 1000 / 1024 / 1024);
-    await remotePeer.DisconnectAsync();
+        // Setting up listener and dialer
+        IListener listener;
+        try
+        {
+            listener = await localPeer.ListenAsync("/ip4/0.0.0.0/tcp/0");
+        }
+        catch (Exception ex)
+        {
+            Console.Error.WriteLine($"Failed to set up listener: {ex.Message}");
+            return;
+        }
+
+        Multiaddress remoteAddr = listener.Address;
+        Console.WriteLine($"Listener is listening on: {remoteAddr}");
+
+        // Dial the Listener from the Remote Peer
+        IRemotePeer connectedPeer;
+        try
+        {
+            connectedPeer = await remotePeer.DialAsync(remoteAddr);
+        }
+        catch (Exception ex)
+        {
+            Console.Error.WriteLine($"Failed to dial the remote peer: {ex.Message}");
+            return;
+        }
+
+        if (connectedPeer == null)
+        {
+            Console.Error.WriteLine("ConnectedPeer is null. Failed to establish connection.");
+            return;
+        }
+
+        Console.WriteLine("Successfully connected to the remote peer.");
+
+        // Start timing the performance test
+        Stopwatch stopwatch = Stopwatch.StartNew();
+        try
+        {
+            // Initiate the PerfProtocol operation
+            await connectedPeer.DialAsync<PerfProtocol>();
+        }
+        catch (Exception ex)
+        {
+            Console.Error.WriteLine($"Error during PerfProtocol DialAsync: {ex.Message}");
+            return;
+        }
+        finally
+        {
+            stopwatch.Stop();
+        }
+
+        TimeSpan elapsedTime = stopwatch.Elapsed;
```

Comment:
So you have stopwatch here, but additionally measure it using that report. I guess this can be optimized